src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceStateOpProcessorDelegate.scala

```diff
@@ -3,47 +3,34 @@ package core.task.tracker.impl
 //scalastyle:off
 import java.time.Clock
 
-import akka.Done
 import akka.actor.ActorRef
 import akka.util.Timeout
 import mesosphere.marathon.core.instance.Instance
 import mesosphere.marathon.core.instance.update.{ InstanceUpdateEffect, InstanceUpdateOperation }
 import mesosphere.marathon.core.task.tracker.impl.InstanceTrackerActor.ForwardTaskOp
-import mesosphere.marathon.core.task.tracker.{ InstanceCreationHandler, InstanceTrackerConfig, TaskStateOpProcessor }
+import mesosphere.marathon.core.task.tracker.{ InstanceTrackerConfig, InstanceStateOpProcessor }
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
 import scala.util.control.NonFatal
 //scalastyle:on
 /**
-  * Implements the [[TaskStateOpProcessor]] trait by sending messages to the TaskTracker actor.
+  * Implements the [[InstanceStateOpProcessor]] trait by sending messages to the TaskTracker actor.
   */
-// TODO(PODS): rename to instance...
-private[tracker] class InstanceCreationHandlerAndUpdaterDelegate(
+private[tracker] class InstanceStateOpProcessorDelegate(
   clock: Clock,
   conf: InstanceTrackerConfig,
   instanceTrackerRef: ActorRef)
-    extends InstanceCreationHandler with TaskStateOpProcessor {
+    extends InstanceStateOpProcessor {
 
   import mesosphere.marathon.core.async.ExecutionContexts.global
 
   private[impl] implicit val timeout: Timeout = conf.internalTaskUpdateRequestTimeout().milliseconds
 
   override def process(stateOp: InstanceUpdateOperation): Future[InstanceUpdateEffect] = {
-    taskUpdate(stateOp.instanceId, stateOp)
-  }
-
-  override def created(taskStateOp: InstanceUpdateOperation): Future[Done] = {
```

Comment:
So this is also implemented somewhere else? Why is this not required anymore?

Reply:
`created` and `terminated` forwarded the call to `process` until now; I've adjusted all call sites to use `process(taskStateOp)` instead, to get rid of the redundant interface

Reply:
Look at the removed impl: it returned the result of `process(taskStateOp).map(_ => Done)`. Now the call sites directly call `process`.

Reply:
Yep, here: https://github.com/mesosphere/marathon/pull/5614/files/d164bb1e4998d6ed044126d7f1e8d8d7b4b966b3#diff-61f68bd4b1b588ea69af252b8377f3efR126
is basically the actual change.